src/adapters/scc.rs

```diff
@@ -0,0 +1,61 @@
+use std::hash::{BuildHasher, Hash};
+use std::sync::Arc;
+
+use super::Value;
+use bustle::*;
+use scc::hash_map::{Entry, HashMap};
+
+#[derive(Clone)]
+pub struct SccMapTable<K, H>(Arc<HashMap<K, Value, H>>)
+where
+    K: Eq + Hash,
+    H: BuildHasher;
+
+impl<K, H> Collection for SccMapTable<K, H>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + Hash + Eq + std::fmt::Debug,
+    H: BuildHasher + Default + Send + Sync + 'static + Clone,
+{
+    type Handle = Self;
+
+    fn with_capacity(capacity: usize) -> Self {
+        Self(Arc::new(HashMap::with_capacity_and_hasher(
+            capacity,
+            H::default(),
+        )))
+    }
+
+    fn pin(&self) -> Self::Handle {
+        self.clone()
+    }
+}
+
+impl<K, H> CollectionHandle for SccMapTable<K, H>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + Hash + Eq + std::fmt::Debug,
+    H: BuildHasher + Default + Send + Sync + 'static + Clone,
+{
+    type Key = K;
+
+    fn get(&mut self, key: &Self::Key) -> bool {
+        self.0.get(key).is_some()
```

Comment:
Hi, I'm one of the authors of the `scc` crate, and just happened to see this PR :-) Note that `get` acquires exclusive ownership of the entry for _modification_, so it is not quite optimal here - I'd suggest you use `read` here.

Reply:
Resolved. Maybe get should be named `get_for_update` like in sql?

Reply:
![ReadHeavy std throughput](https://github.com/xacrimon/conc-map-bench/assets/32823131/b29cce03-864e-43d3-a8ae-7d4d5aff656d)
![ReadHeavy std throughput](https://github.com/xacrimon/conc-map-bench/assets/32823131/dcb59d13-85b5-4fee-ac37-ca7fea70b93e)
@wvwwvwwv not a drammatic change

Reply:
This proves how correctly/meticulously the benchmark suite has been written: keys must be very uniformly distributed.